runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectRunnerTest.java

```diff
@@ -616,6 +649,186 @@ public void processElement(ProcessContext c) {
     p.run();
   }
 
+  /**
+   * Test running of {@link Pipeline} which has two {@link POutput POutputs} and finishing the first
+   * one triggers data being fed into the second one.
+   */
+  @Test(timeout = 10000)
+  public void testTwoPOutputsInPipelineWithCascade() throws InterruptedException {
```

Comment:
This seems like a `@Category(ValidatesRunner.class)` test for use in either `CombineTest` or `GroupByKeyTest`.

Reply:
I'm afraid that could cause trouble. This test uses a (little hackish) class `StaticQueue`, that will not work well outside single JVM. That probably will not be problem for many runners, that use a local version of distributed runner for tests (spark, flink), but might definitely cause troubles for fully distributed runners (dataflow). Because the tooling in the test works only in local mode, I'd prefer to keep it for DirectRunner only.

Reply:
Got it. Makes sense.